classes/class-pmpro-discount-code-list-table.php

```diff
@@ -0,0 +1,516 @@
+		do_action( 'pmpro_discountcodes_extra_cols_header' );
+		$extra_cols = ob_get_clean();
+		preg_match_all( '/<th>(.*?)<\/th>/s', $extra_cols, $matches );
+		$custom_field_num = 0;
+		foreach ( $matches[1] as $match ) {
+			$columns[ 'custom_field_' . $custom_field_num ] = $match;
+			$custom_field_num++;
+		}
+
+		// Shortcut for editing columns in default discount code list location.
+		$current_screen = get_current_screen();
+		if ( ! empty( $current_screen ) && strpos( $current_screen->id, "pmpro-discountcodes" ) !== false ) {
+			$columns = apply_filters( 'pmpro_manage_memberslist_columns', $columns );
+		}
+
+
+		return $columns;
+	}
+
+	/**
+	 * Define which columns are hidden
+	 *
+	 * @return Array
+	 */
+	public function get_hidden_columns() {
+		
+		return array();
+		
+	}
+
+	/**
+	 * Get a list of sortable columns. The format is:
+	 * 'internal-name' => 'orderby'
+	 * or
+	 * 'internal-name' => array( 'orderby', true )
+	 *
+	 * The second format will make the initial sorting order be descending
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	protected function get_sortable_columns() {
+		/**
+		 * actual sorting still needs to be done by prepare_items.
+		 * specify which columns should have the sort icon.
+		 *
+		 * key => value
+		 * column name_in_list_table => columnname in the db
+		 */
+		$sortable_columns = array(
+			'id' => array( 'id', false ),
+			'discount_code' => array( 'discount_code', false ),
+			'starts' => array( 'starts', false ),
+			'expires' => array( 'expires', false ),
+			'uses' => array( 'uses', false ),
+		);
+		// var_dump($sortable_columns);
+		return $sortable_columns;
+	}
+
+	/**
+	 * Text displayed when no user data is available
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function no_items() {
+			
+		esc_html_e( 'No discount codes found.', 'paid-memberships-pro' );
+
+	}
+
+	/**
+	 * Get the table data
+	 *
+	 * @return Array|integer if $count parameter = true
+	 */
+	private function sql_table_data( $count = false ) {
+
+		global $wpdb;
+
+		if( isset( $_REQUEST['s'] ) ) {
+			$s = trim( sanitize_text_field( $_REQUEST['s'] ) );
+		} else {
+			$s = '';
+		}
+
+		// some vars for pagination
+		if(isset($_REQUEST['paged']))
+			$pn = intval($_REQUEST['paged']);
+		else
+			$pn = 1;
+
+		$limit = $this->get_items_per_page( 'discount_codes_per_page' );
+
+		$end = $pn * $limit;
+		$start = $end - $limit;
+
+		if ( $count ) {
+			$sqlQuery = "SELECT COUNT( DISTINCT id ) FROM $wpdb->pmpro_discount_codes ";
+		} else {
+			$sqlQuery = "SELECT SQL_CALC_FOUND_ROWS *, UNIX_TIMESTAMP(CONVERT_TZ(starts, '+00:00', @@global.time_zone)) as starts, UNIX_TIMESTAMP(CONVERT_TZ(expires, '+00:00', @@global.time_zone)) as expires FROM $wpdb->pmpro_discount_codes ";
+		}
+
+		if ( ! empty( $s ) ) {
+			$sqlQuery .= "WHERE code LIKE '%" . esc_sql( $s ) . "%' ";
+		}
+
+		if ( ! $count ) {
+
+			if( isset( $_REQUEST['orderby'] ) ) {
+				$orderby = sanitize_text_field( $_REQUEST['orderby'] );
```

Comment:
Just taking a note to start a discussion. When sorting by `uses`, the sort goes by the number of total uses, not the number of times that the code has already been used.

The weird thing here is that when a code has 0 uses, it is considered unlimited. This means that when sorting from lowest uses to highest, `unlimited` will be first.

Reply:
This wasn't working properly, it should have been sorting by used instead. Pushed a commit further down that fixes this

Reply:
Just tested this PR again on my dev site and it still appears to be sorting based on `uses` instead of `used`. Can you check if you are seeing the same behavior on your site?